src/Navigation.re

```diff
@@ -14,6 +14,25 @@ external navigateWithParams: (t, string, Js.t({..})) => unit = "navigate";
 [@bs.send] external goBackToRoute: (t, string) => unit = "";
 [@bs.send] external getParam: (t, string) => Js.nullable('a) = "";
 [@bs.send] external getParamWithDefault: (t, string, 'a) => 'a = "getParam";
+[@bs.send]
+external addListener:
+  (
+    t,
+    string,
+    {
```

Comment:
This should be defined as a separate type (`navigationEventPayload` if we follow the names in the Flow or TypeScript definitions) beforehand.

According to the docs, it should also have a `state` and `lastState` field.